x-pack/legacy/plugins/epm/server/lib/kibana/index_pattern/install.ts

```diff
@@ -5,16 +5,90 @@
  */
 
 import { SavedObjectsClientContract } from 'kibana/server';
+import { Field } from '../../fields/field';
+import { loadFieldsFromYaml } from '../../elasticsearch/template/install';
+import * as Registry from '../../../registry';
 
-export async function installIndexPattern(
-  name: string,
+interface IndexPatternField extends Field {
+  searchable: boolean;
+  aggregatable: boolean;
+  readFromDocValues: boolean;
+}
+export async function installIndexPatterns(
+  pkgkey: string,
   savedObjectsClient: SavedObjectsClientContract
 ) {
-  await savedObjectsClient.create('index-pattern', getData(name));
+  const registryPackageInfo = await Registry.fetchInfo(pkgkey);
+  const fields = await loadFieldsFromYaml(registryPackageInfo);
+  const kibanaIndexPatternFields = makeKibanaIndexPatternFields(fields);
+  // TODO: separate out logs and metrics patterns
+  await savedObjectsClient.create(
+    'index-pattern',
+    getData('logs-metrics', kibanaIndexPatternFields)
+  );
 }
 
-function getData(name: string) {
+function getData(name: string, fields: IndexPatternField[]) {
   return {
     title: name + '-*',
+    fields: JSON.stringify(fields),
   };
 }
+
+const makeKibanaIndexPatternFields = (fields: Field[]): IndexPatternField[] => {
+  const flattenedFields = flattenFields(fields);
+  const transformedFields = transformFields(flattenedFields);
+  return transformedFields;
+};
+
+const transformFields = (fields: Field[]): IndexPatternField[] => {
+  return fields.map(item => {
+    const newItem = { ...item };
+
+    // map this type to field type
+    if (typeMap[item.type]) {
+      newItem.type = typeMap[item.type];
+    }
+
+    // add some temp values
+    return {
+      searchable: false,
+      aggregatable: false,
+      readFromDocValues: true,
+      ...newItem,
+    };
+  });
+};
+
+const flattenFields = (fields: Field[]): Field[] =>
+  fields.reduce<Field[]>((acc, field) => {
+    if (field.fields && field.fields.length) {
+      acc = acc.concat(flattenFields(field.fields));
+    } else {
+      acc.push(field);
+    }
+    return acc;
+  }, []);
+
+interface TypeMap {
+  [key: string]: string;
+}
+const typeMap: TypeMap = {
```

Comment:
Is this copied from somewhere? Just curious because of the alias comment below.

Reply:
A little from here: https://github.com/elastic/beats/blob/586eac17e8c97f335c84759fb2da53932b187d99/libbeat/kibana/fields_transformer.go#L281

Reply:
I think putting this GH comment as a code comment would be helpful. i.e. link to the file we're using to model the values.